configure.ac

```diff
@@ -548,12 +546,41 @@ if test -z "$ENABLE_SOS_TRUE"; then
 	fi
 fi
 
-dnl ldms-python API's
+dnl ldms-python API's and programs
+AX_WITH_PROG([CYTHON], [cython], [no])
```

Comment:
How about using AC_CHECK_PROGS instead? I think we can use that to do something like:

AC_PATH_PROGS([CYTHON], [cython3.4 cython3.6 cython-3.8 cython], [no])

I believe that this macro will test each binary name in the list one by one, and if one is found in PATH, it stops and sets CYTHON to that.

The next thing to try is using AC_PATH_PROGS_FEATURE_CHECK instead of AC_CHECK_PROGS. It looks like that allows you to include a "feature-test", which could be the version tests (using AX_COMPARE_VERSION, for instance). This macro also allows the user to set our CYTHON variable using the CYTHON environment variable, which seems more consistent with how other compilers/linkers/etc are usually set (CC, LD, etc.). It also has an action-if-not found parameter where we can cat AC_MSG_ERROR(). It looks like this macro could wrap up everything into one section.

Reply:
`AC_PATH_PROGS([CYTHON], ...)` does not add `--with-cython` option to the configure script. The reason `AX_WITH_PROGS` is used is to so that users can specify the path to cython binary using the `--with-cython` option.

Reply:
I don't think that --with-cython is necessarily the right way to do that. We don't do "--with-cc=gcc8932", we do "CC=gcc8932". Likewise, we should probably have "CYTHON=" instead of "--with-cython". The AC_PATH_PROGS_FEATURE_CHECK seems to support the "CYTHON=" method.